executor/analyze_test.go

```diff
@@ -2986,3 +2868,429 @@ func TestShowAanalyzeStatusJobInfo(t *testing.T) {
+	h := dom.StatsHandle()
+	oriLease := h.Lease()
+	h.SetLease(1)
+	defer func() {
+		h.SetLease(oriLease)
+	}()
+	is := dom.InfoSchema()
+	table, err := is.TableByName(model.NewCIStr("test"), model.NewCIStr("t"))
+	require.NoError(t, err)
+	tableInfo := table.Meta()
+	pi := tableInfo.GetPartitionInfo()
+	require.NotNil(t, pi)
+
+	// generate old partition stats
+	tk.MustExec("set global tidb_persist_analyze_options = false")
+	tk.MustExec("set @@session.tidb_partition_prune_mode = 'static'")
+	tk.MustExec("analyze table t partition p0 columns a,c with 1 topn, 3 buckets")
+	tk.MustQuery("select * from t where a > 1 and b > 1 and c > 1 and d > 1")
+	require.NoError(t, h.LoadNeededHistograms())
+	p0 := h.GetPartitionStats(tableInfo, pi.Definitions[0].ID)
+	require.Equal(t, 3, len(p0.Columns[tableInfo.Columns[2].ID].Buckets))
+
+	// analyze partition with existing stats of other partitions
+	tk.MustExec("set @@session.tidb_partition_prune_mode = 'dynamic'")
+	tk.MustExec("analyze table t partition p1 columns a,b,d with 1 topn, 3 buckets")
+	tk.MustQuery("show warnings").Sort().Check(testkit.Rows(
+		"Note 1105 Analyze use auto adjusted sample rate 1.000000 for table test.t's partition p1",
+		"Warning 8244 Build table: `t` global-level stats failed due to missing partition-level column stats, please run analyze table to refresh columns of all partitions",
+	))
+
+	// insert table-level persisted analyze options
+	tk.MustExec("insert into mysql.analyze_options values (?,?,?,?,?,?,?)", tableInfo.ID, 0, 0, 2, 2, "DEFAULT", "")
+	tk.MustExec("analyze table t partition p1 columns a,b,d with 1 topn, 3 buckets")
+	tk.MustQuery("show warnings").Sort().Check(testkit.Rows(
+		"Note 1105 Analyze use auto adjusted sample rate 1.000000 for table test.t's partition p1",
+		"Warning 8244 Build table: `t` global-level stats failed due to missing partition-level column stats, please run analyze table to refresh columns of all partitions",
+	))
+
+	// insert partition-level persisted analyze options
+	tk.MustExec("insert into mysql.analyze_options values (?,?,?,?,?,?,?)", pi.Definitions[0].ID, 0, 0, 2, 2, "DEFAULT", "")
+	tk.MustExec("analyze table t partition p1 columns a,b,d with 1 topn, 3 buckets")
+	tk.MustQuery("show warnings").Sort().Check(testkit.Rows(
+		"Note 1105 Analyze use auto adjusted sample rate 1.000000 for table test.t's partition p1",
+		"Warning 8244 Build table: `t` global-level stats failed due to missing partition-level column stats, please run analyze table to refresh columns of all partitions",
+	))
+	tk.MustQuery("select * from t where a > 1 and b > 1 and c > 1 and d > 1")
+	require.NoError(t, h.LoadNeededHistograms())
+	tbl := h.GetTableStats(tableInfo)
+	require.Equal(t, 0, len(tbl.Columns))
+
+	tk.MustExec("analyze table t partition p1 columns a,b,c with 1 topn, 3 buckets")
+	tk.MustQuery("show warnings").Sort().Check(testkit.Rows(
+		"Note 1105 Analyze use auto adjusted sample rate 1.000000 for table test.t's partition p1",
```

Comment:
It seems that `p0` doesn't have stats of column `b`. But we don't see `ErrPartitionColumnStatsMissing` here.